samples/grid/griddemo.cpp

```diff
@@ -2850,3 +2859,86 @@ void GridFrame::ShowRow( wxCommandEvent& WXUNUSED(event) )
 {
     grid->ShowRow(1);
 }
+
+namespace
+{
+
+// Toggle status of either checkered or coloured cells in the grid.
+// Note that, for shared attribute testing purposes, with checkered cells
+// a cell's complete attribute is destructively overwritten (except merged
+// cells which are skipped). While with coloured cells only the background
+// colour changes.
+void ToggleGridCells(wxGrid* grid, bool useCheckered)
+{
+    static bool _apply[2] = { };
+    const bool apply = _apply[useCheckered] = !_apply[useCheckered];
```

Comment:
Famous last-minute changes, I must have been blinded by introducing `apply` and not repeating `apply[useCheckered]` for a whole two times or something. Just before it was:

```cpp
static bool apply[2] = { };
apply[useCheckered] = !apply[useCheckered];
```

And what if `useCheckered` (/`effectIndex`) is `unsigned` (or rather `unsigned int` in wx) instead?

Then again I guess two statics could just be used, I just felt like it could discourage the addition of, for whatever reason, another effect somewhat.

I suppose the single underscore avoiding is a style thing (it's reserved for globals was my conclusion when checking). I sometimes use the ugly underscore exactly to discourage using such variables further in code, as here convenience variable `apply` should only have been used instead.